src/System.Diagnostics.Process/src/System/Collections/Specialized/StringDictionaryWithComparer.cs

```diff
@@ -0,0 +1,170 @@
+    is private and we don't want to make that field protected. */
+
+    [Serializable]
+    internal class StringDictionaryWithComparer : StringDictionary {
+
+        private Dictionary<string, string> _contents;
+        private readonly object _syncRoot = new Object();
+
+        public StringDictionaryWithComparer() : this(StringComparer.OrdinalIgnoreCase) { }
+
+        public StringDictionaryWithComparer(IEqualityComparer<string> comparer)
+        {
+            _contents = new Dictionary<string, string>(comparer);         
+        }
+
+        public override int Count
+        {
+            get
+            {
+                return _contents.Count;
+            }
+        }
+        
+        public override string this[string key]
+        {
+            get
+            {
+                if(key == null)
+                {
+                    throw new ArgumentNullException(nameof(key));
+                }
+
+                return _contents[key];
+            }
+            set 
+            {
+                if(key == null)
+                {
+                    throw new ArgumentNullException(nameof(key));
+                }
+
+                _contents[key] = value;
+            }
+        }
+
+        public override ICollection Keys
+        {
+            get
+            {
+                return _contents.Keys;
+            }
+        }
+
+        public override ICollection Values
+        {
+            get
+            {
+                return _contents.Values;
+            }
+        }
+
+        public override bool IsSynchronized
+        {
+            get
+            {
+                return false;
+            }
+        }
+
+        public override object SyncRoot
+        {
+            get
+            {
+                return _syncRoot;
+            }
+        }
+
+        public override void Add(string key, string value)
+        {
+            if(key == null)
+            {
+                throw new ArgumentNullException(nameof(key));
+            }
+
+            _contents.Add(key, value);
+        }
+
+        public override void Clear()
+        {
+            _contents.Clear();
+        }
+
+        public override bool ContainsKey(string key)
+        {
+            if(key == null) 
+            {
+                throw new ArgumentNullException(nameof(key));
+            }
+
+            return _contents.ContainsKey(key);
+        }
+
+        public override bool ContainsValue(string value)
+        {
+            return _contents.ContainsValue(value);
+        }
+
+        //Using a list so that List.CopyTo handles the right exceptions.
+        public override void CopyTo(Array array, int index)
+        {
+
+            if(array == null)
+            {
+                throw new ArgumentNullException(nameof(array));
+            }
+
+            if(array as KeyValuePair<string,string>[] == null)
```

Comment:
Nit: This would be better written as:
```C#
var kvpArray = array as KeyValuePair<string,string>[];
if (kvpArray == null)
{
    throw new ArrayTypeMismatchException();
}
```
and then down below where it does:
```C#
list.CopyTo(array as KeyValuePair<string, string>[], index);
```
would instead be:
```C#
list.CopyTo(kvpArray, index);
```